pkg/debuginfo/manager.go

```diff
@@ -51,6 +52,8 @@ type Manager struct {
 	stripDebuginfos bool
 	tempDir         string
 
+	// hashCache caches ELF hashes (buildID is a key).
+	hashCache                         *sync.Map
```

Comment:
With a `sync.Map` the benchmark shows one less allocation comparing to `cache.Cache`.